source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,32 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+#include <stdio.h>
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_collection_t *collection;
+   mongoc_client_t *client;
+   int64_t count;
+
+   mongoc_init ();
+
+   client = mongoc_client_new ("mongodb+srv://<cluster-url>/test?authSource=$external&tlsCertificateKeyFile=/etc/certs/mongodb/client.pem&retryWrites=true&w=majority&authMechanism=MONGODB-X509");
+
+   collection = mongoc_client_get_collection (client, "testDB", "testCol");
+   bson_t filter = BSON_INITIALIZER;
+   count = mongoc_collection_count_documents (collection, &filter, NULL, NULL, NULL, NULL);
```

Comment:
http://mongoc.org/libmongoc/1.14.0/mongoc_collection_count_documents.html#example already has a good example for passing a `bson_errot_t` address as the last argument and using that for printing an error message. I'd suggest you just borrow that syntax, which would also change the conditional to `count < 0`.

Reply:
Fixed